### PR TITLE
Create Folders when moving files

### PR DESCRIPTION
**Pull request by rummatee**
This Pull Request intends to solve Issue #7629 .
It adds a button to the file picker next to the breadscrumbs. When clicking on this button it opens a popup that asks for the folders name. After submitting the form, it closes the popup and reloads the file picker's content.
There is one small bug, that I couldn't fix: OC.hideMenus(); prevents the file picker from reacting to keyboard button presses until it is clicked again. So after creating a new folder the enter and escape key will only work again after the user clicked the file picker again.

Please review @MariusBluem @jancborchardt

### CODE REVIEW

**Comment by jancborchardt** [comment]
Nice! This seems to work well, only some feedback:
- [ ] When navigating with the keyboard (via tab to switch elements and enter to "click"), pressing enter on the "New folder" button closes the dialog
- [ ] The distance of the button to the end of the breadcrumbs is a bit much. Should be the same as in the Files app
   screenshot from 2018-12-20 22-55-33
   move copy dialog
- [ ] When choosing a new folder name and pressing enter to confirm, the folder shows up twice or more initially (of course only exists once):
screenshot from 2018-12-20 22-55-17
- [ ] When you create a new folder, and then another one, the input placeholder should be reset to "New folder"

What do you think @rummatee? :)

**Comment by jancborchardt** [comment]
Also @MariusBluem @fermulator for review since you were active in related issues. :)

**Comment by danielkesselberg** [comment]
image

Button looks a bit off to me (Chromium 70+). Maybe related to your last changes because @jancborchardt screenshots looks ok or a local issue on my side.

**Review by kesselb — APPROVED** [review]
Good stuff :+1: Please align the button :sunglasses:

**Comment by rummatee** [comment]
> Button looks a bit off to me (Chromium 70+). Maybe related to your last changes because @jancborchardt screenshots looks ok or a local issue on my side.

It seems to be related to Chrome. In Firefox it showed the position correctly. However my last commit should fix the position for both chrome and Firefox

**Comment by jancborchardt** [comment]
@nextcloud/designers can we have another review? :)

**Comment by violoncelloCH** [comment]
@rummatee Could this be rebased on current master? My dev setup instance complains that downgrading to 15.0.5 is unsupported, so I don't know how I should test this actually.

**Comment by violoncelloCH** [comment]
thank you @danielkesselberg! that finally worked...

File: core/js/jquery.ocdialog.js [hunk 1]
```diff
@@ -54,6 +56,11 @@
 				// Enter
 				if(event.keyCode === 13) {
 					event.stopImmediatePropagation();
+					if (self.enterCallback != null) {
```
Comment (violoncelloCH) [inline core/js/jquery.ocdialog.js]: according to the code style guideline you should use `!==` here (https://docs.nextcloud.com/server/15/developer_manual/general/codingguidelines.html#operators)

File: core/js/oc-dialogs.js [hunk 2]
```diff
@@ -251,6 +255,86 @@ var OCdialogs = {
...
+						self._fillFilePicker(self.$filePicker.data('path') );
+					});
+					OC.hideMenus();
+					self.$filePicker.ocdialog('unsetEnterCallback');
+					self.$filePicker.click();
+					$input.val(newText);
+				}
+			});
+			$input.keypress(function(event) {
+				if (event.keyCode == 13 || event.which == 13) {
```
Comment (violoncelloCH) [inline core/js/oc-dialogs.js]: same here (https://docs.nextcloud.com/server/15/developer_manual/general/codingguidelines.html#operators)

**Review by violoncelloCH — CHANGES_REQUESTED** [review]
otherwise it looks and works great!

**Comment by violoncelloCH** [comment]
Also, if you create a folder in this "choose target folder" dialog, you normally want to place the file inside of this folder. So wouldn't it make sense to directly enter into the just created folder? What do you think @jancborchardt ?

**Comment by jancborchardt** [comment]
> So wouldn't it make sense to directly enter into the just created folder?

Yeah, that would be a nice addition! :)

**Comment by danielkesselberg** [comment]
Some tests are failing. Could you rebase this branch onto master?

**Review by violoncelloCH — CHANGES_REQUESTED** [review]
perfect from my view otherwise! :+1:

**Review by violoncelloCH — APPROVED** [review]
:+1:

**Comment by jancborchardt** [comment]
What about the Drone failure here, does that need to be fixed?

**Comment by danielkesselberg** [comment]
Another rebase should fix the litmus test. Not sure about the other tests. Restarted the build just in case.

**Comment by danielkesselberg** [comment]
Another test fails now. I don't see how these changes are related to the failing test.

**Comment by jancborchardt** [comment]
@MorrisJobke @rullzer could you check what it is about the failing tests? :man_singer:

**Comment by MorrisJobke** [comment]
> @MorrisJobke @rullzer could you check what it is about the failing tests? 👨‍🎤

Those are all hiccups of our test suite 🙈 We should maybe get rid of them, because they are a major annoyance. cc @juliushaertl @danxuliu @rullzer

**Comment by jancborchardt** [comment]
@rummatee please avoid force-pushing by the way as it makes it difficult to see which changes occured since the last time someone checked. :)

**Comment by jancborchardt** [comment]
Ah ok, now the test runs through, that’s why you force-pushed. Sorry and nevermind! :)

**Comment by rummatee** [comment]
> @rummatee please avoid force-pushing by the way as it makes it difficult to see which changes occured since the last time someone checked. :)

I'm supposed to rebase the pull request, right? Is there a different way to do it?

**Comment by MorrisJobke** [comment]
> I'm supposed to rebase the pull request, right? Is there a different way to do it?

Typically if changes are added, then we do fixup commits and then a final rebase with `--interactive --autosquash` so that they will get squashed into one or multiple commits. That makes it usually easier. In this case if you want to get master stuff in the rebase is still fine IMO.

**Comment by jancborchardt** [comment]
And @rummatee @danielkesselberg do you know about our next Contributor Week in Stuttgart, March 11–15? Contributors like you can apply for travel/accomodation support, find more info in the blog post at https://nextcloud.com/blog/the-contributor-week-is-over-time-to-plan-for-the-next-one/ :slightly_smiling_face:

**Pull request merged**